### DISCUSSION TITLE
Minor Keyword Issue

### DISCUSSION THREAD

**Discussion by mosqu-ito** [SpenceKonde/megaTinyCore#720] [Q&A] [answered]
Hello Sir.
I tried to reach you at your gmail address a week or two ago; don't know if I succeeded.  But if I were a junk mail filter, I would probably shove me straight into the trash too.  Actually I had only written to say "Thank You" for this core and to let you know that your UPDI instructions had worked for me right out of the box.
After some tinkering with my 3224s, I've started playing with some 3226s I got just recently.  I'm having a world of fun, thanks to you.

Anyway, my big issue...?  It's not orange.  "pinModeFast" I mean.  "digitalReadFast" and "digitalWriteFast" are, and of course "pinMode" is, but not "pinModeFast".

Earth-shattering, n'est ce pas?   :D
Thanks again.

**Comment by SpenceKonde** [comment, marked as answer]
Hmmm, more like mysterious. Because they definitely are syntax highlighted for me, in orange. 

It's achieved via the "dummy" library (megaTinyCore library included with core, which isn't quite a dummy library anymore - it does define a few functions of it's own now, originally it was there just to highlight keywords) keyword.txt file

**Reply by mosqu-ito** [reply]
When I do a <ctrl>F search of the keywords.txt file for "Fast" I find three matches:
openDrainFast	KEYWORD2
digitalWriteFast	KEYWORD2
digitalReadFast	KEYWORD2
I do not find "pinModeFast".
Presumably your keywords file _does_ have that entry.
So... how did I end up with a different file?
That _would_ seem to be a mystery.
Thanks for the reply, Sir, I know your time must be in great demand.

**Reply by mosqu-ito** [reply]
I had typed "control F" with angle braces but I guess that got taken as some erroneous html.
Anyway, however I came to have my deficient file, you've certainly given me all I need to correct it,
so I should mark my query as answered.  Thanks!

**Comment by SpenceKonde** [comment]
Yeah, pinModeFast was missing from keywords.txt I only checked a couple of
the others, I didn't realize it was specifically pinModeFast that was
broken.

On Sat, Jun 4, 2022 at 4:04 PM mosqu-ito ***@***.***> wrote:

> I had typed "control F" with angle braces but I guess that got taken as
> some erroneous html.
> Anyway, however I came to have my deficient file, you certainly given me
> all I need to correct it,
> so I should mark my query as answered. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/megaTinyCore/discussions/720#discussioncomment-2884262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW7IBIN7OMVUGJTP2HTVNOZEXANCNFSM5X2QDTGQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***
